R/DataDiscoveryRetrieval.R

```diff
@@ -324,146 +351,176 @@ TADA_DataRetrieval <- function(startDate = "null",
         "American Indian Reservations",
         "Off-reservation Trust Lands",
-        "Oklahoma Tribal Statistical Areas"#,
-        # "Virginia Federally Recognized Tribes"
+        "Oklahoma Tribal Statistical Areas"
       )
-      ){
-        
+      ) {
         # Get the relevant url
-        aoi_sf <- filter(map_service_urls,
-                         tribal_area == tribal_area_type)$url %>%
+        aoi_sf <- dplyr::filter(
+          map_service_urls,
+          tribal_area == tribal_area_type
+        )$url %>%
           # Pull data
           arcgislayers::arc_open() %>%
           # Return sf
           arcgislayers::arc_select() %>%
           # If a value provided, then filter
-          {if ((tribe_name_parcel != "null") & (tribe_name_parcel != "null")) {
-            filter(., TRIBE_NAME %in% tribe_name_parcel)
-          } else {
-            .
-          }}
-        
+          {
+            if (all(tribe_name_parcel != "null")) {
+              dplyr::filter(., TRIBE_NAME %in% tribe_name_parcel)
+            } else {
+              .
+            }
+          }
+
         # Otherwise filter by PARCEL_NO (Note that values in this col are not unique)
-      } else if(tribal_area_type == "Alaska Native Allotments"){
-        
-        aoi_sf <- filter(map_service_urls,
-                         tribal_area == tribal_area_type)$url %>%
+      } else if (tribal_area_type == "Alaska Native Allotments") {
+        aoi_sf <- dplyr::filter(
+          map_service_urls,
+          tribal_area == tribal_area_type
+        )$url %>%
           arcgislayers::arc_open() %>%
           arcgislayers::arc_select() %>%
-          {if ((tribe_name_parcel != "null") & (tribe_name_parcel != "null")) {
-            filter(., PARCEL_NO %in% tribe_name_parcel)
-          } else {
-            .
-          }}
-        
+          {
+            if (all(tribe_name_parcel != "null")) {
+              dplyr::filter(., PARCEL_NO %in% tribe_name_parcel)
+            } else {
+              .
+            }
+          }
       } else {
-        stop("Tribal area type not recognized. Refer to TADA_TribalOptions() for query options.")
+        stop("Tribal area type or tribal name parcel not recognized. Refer to TADA_TribalOptions() for query options.")
       }
-      
     }
-    
+
     # Check and/or fix geometry
     aoi_sf <- sf::st_make_valid(aoi_sf)
-    
+
     # Match CRS
-    if(sf::st_crs(aoi_sf) != 4326){
+    if (sf::st_crs(aoi_sf) != 4326) {
       aoi_sf <- sf::st_transform(aoi_sf, crs = 4326)
     }
-    
+
     # Get bbox of the sf object
     input_bbox <- sf::st_bbox(aoi_sf)
-    
+
     # Query info on available data within the bbox
-    bbox_avail <- suppressMessages(
-      dataRetrieval::whatWQPdata(
-        WQPquery,
-        bBox = c(input_bbox$xmin, input_bbox$ymin, input_bbox$xmax, input_bbox$ymax)
-      )
+
+    # Try getting WQP info
+    message("Checking what data is available. This may take a moment.")
```

Comment:
```suggestion
    message("Checking for available data. This may take a moment.")
```

Maybe move this up to the very top of the function right after the sanity checks?